packages/core/app-extensions/src/reports/sagas.js

```diff
@@ -22,22 +24,46 @@ export function* loadReport({payload: {reportIds, entityName, scope}}) {
 
   const query = {
     where: reportConditions,
-    paths: 'unique_id,label,ignore_selection'
+    paths: 'unique_id,label,ignore_selection,relOutput_template.relReport_file_format.unique_id'
   }
 
   const result = yield call(rest.fetchEntities, 'Report', query, {method: 'GET'})
 
-  const reports = result.map(r => ({
+  const reports = yield all(
+    result.map(entity => api.getFlattenEntity(entity)).map(report => call(buildReportDefinition, report))
+  )
+
+  yield put(actions.setReports(reports))
+}
+
+function* buildReportDefinition(report) {
+  return {
     actionType: 'report',
     componentType: 'report',
-    confirmationThreshold: r.paths.ignore_selection.value ? null : DEFAULT_CONFIRMATION_THRESHOLD,
-    id: r.paths.unique_id.value,
-    label: r.paths.label.value,
-    reportId: r.paths.unique_id.value,
-    showConfirmation: !r.paths.ignore_selection.value
-  }))
+    confirmationThreshold: report.ignore_selection ? null : DEFAULT_CONFIRMATION_THRESHOLD,
+    id: report.unique_id,
+    label: report.label,
+    reportId: report.unique_id,
+    showConfirmation: !report.ignore_selection,
+    icon: yield call(getReportIcon, report)
+  }
+}
 
-  yield put(actions.setReports(reports))
+function* getReportIcon(report) {
+  const reportIcons = yield call(loadReportIcons)
+  const fileType = report['relOutput_template.relReport_file_format.unique_id']
+  return reportIcons[fileType]
```

Comment:
if a file type has no icon defined in the backend should there be a generic fallback icon or just no icon?

Reply:
I'd go with there being no icon, since that is what ReportActionFormInterceptor in the backend does currently.